### PR TITLE
Warn to seed the database when no Developer records in development

### PR DESCRIPTION
**Pull request by bensheldon**
Connects to #472; broken out of #484 .

Adds a warning when in the Rails Development environment and the database has no Developer records to seed the database.

Screen Shot 2022-06-15 at 5 52 52 PM

- [x] My code contains tests covering the code I modified
- [x] I linted and tested the project with `bin/check`
- [x] I added significant changes and product updates to the [changelog](CHANGELOG.md) -- I think #484 will cover it.

### CODE REVIEW

File: app/views/developers/index.html.erb [hunk 1]
```diff
@@ -41,6 +41,7 @@
 
     <!-- Paginated list of developers -->
     <div class="col-span-12 md:col-span-9">
+      <%= render Developers::UnseededWarningComponent.new(seedable: Rails.env.development?) %>
```
Comment (joemasilotti) [inline app/views/developers/index.html.erb:44]: Let's move this to `app/views/shared/_main.html.erb:2` so it shows up on every page.

File: test/components/developers/unseeded_warning_component_test.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+module Developers
+  class UnseededWarningComponentTest < ViewComponent::TestCase
+    test "hidden when unseedable" do
+      render_inline UnseededWarningComponent.new(seedable: false)
+      assert_no_text "Run bin/rails db:seed"
+    end
+
+    test "hidden when seedable but developers exist" do
+      render_inline UnseededWarningComponent.new(seedable: true)
+      assert_no_text("Run bin/rails db:seed")
```
Comment (joemasilotti) [inline test/components/developers/unseeded_warning_component_test.rb:12]: Let's consolidate using parenthesis or not for these assertions.

**Review by joemasilotti — CHANGES_REQUESTED** [review]
Two minor requests and we are good to merge!

File: test/components/developers/unseeded_warning_component_test.rb (hunk 2 above)
Comment (bensheldon) [reply test/components/developers/unseeded_warning_component_test.rb]: Nice catch. I removed them to be consistent 👍

File: app/views/developers/index.html.erb (hunk 1 above)
Comment (bensheldon) [reply app/views/developers/index.html.erb]: 👍

**Comment by bensheldon** [comment]
@joemasilotti I've updated this PR based on your feedback. Looking sharp ✨ 

Screen Shot 2022-06-15 at 5 52 52 PM

**Comment by joemasilotti** [comment]
Excellent - thanks for the contribution!

**Pull request merged**